gemini/sample-apps/fashion-trends-prediction/app/utilities.py

```diff
@@ -0,0 +1,187 @@
+                }
+            }
+            window.addEventListener("load", function() {
+                attempt_nav_page("%s", new Date(), %d);
+            });
+        </script>
+    """ % (
+        page_name,
+        timeout_secs,
+    )
+    html(nav_script)
+
+
+# To remove background from an image
+def removeBg(url, userName):
+    """Removes the background from an image.
+
+    Args:
+        url (str): The URL of the image to remove the background from.
+        userName (str): The name of the user who uploaded the image.
+
+    Returns:
+        str: The path to the image with the background removed.
+
+    """
+    input_path = url
+    actual_img_path = "" + userName + "$.png"
+    urllib.request.urlretrieve(input_path, actual_img_path)
+
+    # Store path of the output image in the variable output_path
+    output_path = "" + userName + "#.png"
+
+    # Processing the image
+    input = ImagePIL.open(actual_img_path)
+
+    # Removing the background from the given Image
+    output = remove(input)
+
+    # Saving the image in the given path
+    output.save(output_path)
+
+    return actual_img_path, output_path
+
+
+def get_base64_of_bin_file(png_file):
+    """Converts a binary file to a base64 encoded string.
+
+    Args:
+        png_file (str): The path to the binary file.
+
+    Returns:
+        str: The base64 encoded string of the binary file.
+
+    """
+    with open(png_file, "rb") as f:
+        data = f.read()
+    return base64.b64encode(data).decode()
+
+
+def render_svg(svg):
+    """Renders an SVG string.
+
+    Args:
+        svg (str): The SVG string to render.
+
+    """
+    """Renders the given svg string."""
+    b64 = base64.b64encode(svg.encode("utf-8")).decode("utf-8")
+    html = r'<img src="data:image/svg+xml;base64,%s"/>' % b64
+    st.write(html, unsafe_allow_html=True)
+
+
+def build_markup_for_logo(
+    png_file,
+    background_position="50% 10%",
+    margin_top="5%",
+    image_width="80%",
+    image_height="",
+):
+    """Builds the HTML markup for a logo.
+
+    Args:
+        png_file (str): The path to the PNG file of the logo.
+        background_position (str, optional): The background position of the logo. Defaults to "50% 10%".
+        margin_top (str, optional): The margin top of the logo. Defaults to "5%".
+        image_width (str, optional): The width of the logo. Defaults to "80%".
+        image_height (str, optional): The height of the logo. Defaults to "".
+
+    Returns:
+        str: The HTML markup for the logo.
+
+    """
+    binary_string = get_base64_of_bin_file(png_file)
+    return """
+            <style>
+                [data-testid="stSidebarNav"] {
+                    background-image: url("data:image/png;base64,%s");
+                    background-repeat: no-repeat;
+                    background-position: %s;
+                    margin-top: %s;
+                    background-size: %s %s;
+                }
+            </style>
```

Comment:
Not super familiar with streamlit, but is there a more modular way to create these kinds of elements? Could they be brought out into HTML files or something?